pkg/hosts/hosts.go

```diff
@@ -113,7 +113,23 @@ func (h *Hosts) Add(ipRaw string, hosts []string) error {
 		hfl.Raw = h.File.RenderHostsFileLine(newLineNum)
 
 	} else {
-		line.Hostnames = append(line.Hostnames, hostEntries...)
+		var hostToAdd []string
+		for _, hostName := range hostEntries {
+			// check that new host not present in this line
```

Comment:
this duplicates the code in `Contains`, I'd add a `lineContains(line, host) bool` helper, this also removes the need for the code comment